Update golangci-lint to v1.50.1 in the golang-lint convention

mjlshen opened the pull request:
Go 1.19 is out and about, but v1.48.0 is the oldest to support it - https://github.com/golangci/golangci-lint/releases

Might as well get the latest? Also found some cruft with unused OPM/GRPCURL versions being specified

There have since been some deprecated checks:
```
WARN [runner] The linter 'varcheck' is deprecated (since v1.49.0) due to: The owner seems to have abandoned the linter.  Replaced by unused. 
WARN [runner] The linter 'structcheck' is deprecated (since v1.49.0) due to: The owner seems to have abandoned the linter.  Replaced by unused. 
WARN [runner] The linter 'deadcode' is deprecated (since v1.49.0) due to: The owner seems to have abandoned the linter.  Replaced by unused. 
WARN [linters_context] structcheck is disabled because of generics. You can track the evolution of the generics support by following the https://github.com/golangci/golangci-lint/issues/2649. 
```

So I have disabled those as well.

2uasimojo commented:
That said, this convention looks like it's only used in osd-cluster-ready, osd-network-verifier, and operator-custom-metrics; so maybe use one of those instead.

Out of curiosity, why not bump this in the other conventions as well?

mjlshen commented:
> This is nice. What would be cool is if you tested this locally with a large, complex consumer (aws-account-operator?) to make sure it doesn't raise issues of such quantity/nature that they would be impractical or slow/heavy to fix. I.e. we wouldn't want this bp update to block subsequent ones that might be more important.

> Out of curiosity, why not bump this in the other conventions as well?

Related reasons :D My intention is to upgrade osd-network-verifier to Go 1.19, but this is the minimum blocker https://github.com/openshift/release/pull/35788 for that. I didn't want to get bogged down with testing the golang-osd-operator convention haha

2uasimojo commented:
Cool cool.

/lgtm

/hold

Unhold whenever you're comfortable this won't unduly burden/block the consumers.

mjlshen commented:
Good thing you had me test it, it works on the individual repos, but there are some deprecated checks now:

```
WARN [runner] The linter 'varcheck' is deprecated (since v1.49.0) due to: The owner seems to have abandoned the linter.  Replaced by unused. 
WARN [runner] The linter 'structcheck' is deprecated (since v1.49.0) due to: The owner seems to have abandoned the linter.  Replaced by unused. 
WARN [runner] The linter 'deadcode' is deprecated (since v1.49.0) due to: The owner seems to have abandoned the linter.  Replaced by unused. 
WARN [linters_context] structcheck is disabled because of generics. You can track the evolution of the generics support by following the https://github.com/golangci/golangci-lint/issues/2649. 
```

So I will disable those shortly.

2uasimojo commented:
/lgtm

mjlshen commented:
/unhold
